server/supernova/sc/sc_osc_handler.cpp

```diff
@@ -1988,9 +1988,13 @@ template <bool realtime> void handle_b_alloc(ReceivedMessage const& msg, endpoin
 
     args >> bufferIndex >> frames;
 
-    if (!args.Eos())
+    float sampleRate = sc_factory->world.mSampleRate;
```

Comment:
After `extract_completion_message`! See corresponding comment in `Buffer.allocMsg`

Also, default to `world.mSampleRate` if the sample rate argument is 0.0.

Reply:
Another note: please follow the style of the other arguments:
```
osc::int32 bufferIndex, frames, channels;
float sampleRate;
...
if (!args.Eos()) {
    args >> sampleRate;
    if (sampleRate == 0.0)
        sampleRate = sc_factory->world.mSampleRate; 
} else {
    sampleRate = sc_factory->world.mSampleRate;
```